file-formats/smtg/type/zob_aff_smtg_v1.intf.abap

```diff
@@ -0,0 +1,156 @@
+interface zob_aff_smtg_v1
+  public.
+
+  " Email Template Metadata --------------------------------------------------------------------------------------------
+  types:
+    "! <p class="shorttext">Administrative Data</p>
+    "! Contains administrative details such as creation and last change information.
+    begin of ty_administrative_data,
+      "! <p class="shorttext">Created By</p>
+      "! Created by
+      "! $showAlways
+      created_by      type string,
+      "! <p class="shorttext">Last Changed By</p>
+      "! last changed by
+      "! $showAlways
+      last_changed_by type string,
+    end of ty_administrative_data.
+
+  types:
+    "! <p class="shorttext">Header</p>
+    "! Contains identifying and descriptive information about the template.
+    begin of ty_template_header,
+      "! <p class="shorttext">Template Name</p>
+      "! Unique identifier for the template
+      "! $required
+      template_id  type smtg_tmpl_name,
+      "! <p class="shorttext">CDS View</p>
+      "! The corresponding CDS view name for the template
+      "! $required
+      cds_view     type smtg_cds_view,
+      "! <p class="shorttext">Description</p>
+      "! Brief description of the template
+      description  type smtg_tmpl_description,
+      "! <p class="shorttext">Predelivered</p>
+      "! Indicator if the template is pre-delivered
+      predelivered type abap_bool,
+    end of ty_template_header.
+
+  types:
+    "! <p class="shorttext">General Information</p>
+    "! Combines template header and administrative data.
+    begin of ty_general_information,
+      "! <p class="shorttext">Template Header</p>
+      "! Contains basic template identifiers and description
+      "! $required
+      template_header     type ty_template_header,
+      "! <p class="shorttext">Administrative Data</p>
+      "! Contains info about template creation and modification
+      "! $required
+      administrative_data type ty_administrative_data,
```

Comment:
since administrative_data can be removed, ty_general_information and template_header can be merged into 1